worker/src/ocall_bridge/ffi/get_ias_socket.rs

```diff
@@ -0,0 +1,63 @@
+/*
+    Copyright 2019 Supercomputing Systems AG
+    Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+use crate::ocall_bridge::bridge_api::{Bridge, RemoteAttestationOCall};
+use log::*;
+use sgx_types::{c_int, sgx_status_t};
+use std::sync::Arc;
+
+#[no_mangle]
+pub extern "C" fn ocall_get_ias_socket(ret_fd: *mut c_int) -> sgx_status_t {
+    get_ias_socket(ret_fd, Bridge::get_ra_api())
+}
+
+fn get_ias_socket(ret_fd: *mut c_int, ra_api: Arc<dyn RemoteAttestationOCall>) -> sgx_status_t {
+    debug!("    Entering ocall_get_ias_socket");
+    let socket = ra_api.get_ias_socket();
+
+    unsafe {
+        *ret_fd = socket;
+    }
+    sgx_status_t::SGX_SUCCESS
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use crate::ocall_bridge::bridge_api::MockRemoteAttestationOCall;
+    use std::sync::Arc;
+
+    #[test]
+    fn get_socket_sets_pointer_result() {
+        let expected_socket = 4321i32;
+
+        let mut ra_ocall_api_mock = MockRemoteAttestationOCall::new();
+        ra_ocall_api_mock
+            .expect_get_ias_socket()
+            .times(1)
+            .return_const(expected_socket);
```

Comment:
Example of how to use the mocking framework:
* Framework generates a new type, with `Mock` prefix and a static `new()` method to construct
* We can setup the return values for methods, using `expect_<method name>()`
** Using `times()` we can define how many times this method is expected to be called